### PR TITLE
feat(input): new password input component

### PR DESCRIPTION
**Pull request by NataliaAlesencova**

### CODE REVIEW

**Comment by NataliaAlesencova** [comment]
so I used yalc to see how this is going to work in discovery, but still seeing same errors for Register and SignIn components. 
cc @hevele-moda

File: src/components/PasswordInput/PasswordInput.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { Input } from '../Field';
+
+import './PasswordInput.scss';
+
+export type PasswordInputProps = {
+  error?: string | boolean | undefined;
+  dataTestId?: string;
+  id: string;
+  placeholder: string;
+};
```
Comment (hevele-moda) [inline src/components/PasswordInput/PasswordInput.tsx:13]: Let's make this accept all props of `Input` and add extra ones if needed, similar to how we're doing it for Button here https://github.com/ModaOperandi/om/blob/master/src/components/Button/Button.tsx#L7-L13

**Comment by hevele-moda** [comment]
> so I used yalc to see how this is going to work in discovery, but still seeing same errors for Register and SignIn components.
> cc @hevele-moda

ok, interesting, what's the error?

**Comment by NataliaAlesencova** [comment]
> > so I used yalc to see how this is going to work in discovery, but still seeing same errors for Register and SignIn components.
> > cc @hevele-moda
> 
> ok, interesting, what's the error?

when using `<PasswordInput />` in SignIn, I'm seeing this 2FA error: 
Screen Shot 2021-09-14 at 11 20 49 AM

when using `<PasswordInput />` in Register, if the field is blank it's supposed to show `'Please enter a password.'` error, but that field doesn't get returned at all, and therefore won't render.

File: src/components/PasswordInput/PasswordInput.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export type PasswordInputProps = {
+  dataTestId?: string;
+} & FieldProps;
+
+export const PasswordInput: React.FC<PasswordInputProps> = ({
+  error,
+  focus,
+  dataTestId,
+  id,
+  placeholder
```
Comment (hevele-moda) [inline src/components/PasswordInput/PasswordInput.tsx:17]: Can you add `...rest` here and pass that down to `Input`?

**Comment by hevele-moda** [comment]
@NataliaAlesencova I think the problem here is probably that the `useForm` hook doesn't pick this input up for some reason. Let's try to investigate that. Let me know if you get stuck and I can help to figure it out.

**Review by hevele-moda — APPROVED** [review]

**Pull request merged**